src/layer/vector/Polyline.js

```diff
@@ -228,3 +223,12 @@ L.Polyline = L.Path.extend({
 L.polyline = function (latlngs, options) {
 	return new L.Polyline(latlngs, options);
 };
+
+L.extend(L.Polyline, {
+
+	_flat: function (latlngs) {
+		// true if it's a flat array of latlngs; false if nested
+		return !L.Util.isArray(latlngs[0]) || (typeof latlngs[0][0] !== 'object' && typeof latlngs[0][0] !== 'undefined');
+	}
+
```

Comment:
No need to use `extend` if we're only adding one function. `L.Polyline._flat = ...`.